.github/actions/surefire-reports/action.yml

```diff
@@ -0,0 +1,41 @@
+name: Archive Surefire reports
+description: It will upload and archive surefire reports per particular test run.
+inputs:
+  job-name:
+    description: 'The name of the particular job run.'
+    required: true
+  release-branches:
+    description: 'List of all related release branches (in JSON format)'
+    required: false
+    default: '["refs/heads/release/22.0"]'
+  path-pattern:
+    description: 'The path pattern from where to get the surefire reports.'
+    required: false
+    default: |
+      **/target/surefire-reports/TEST*.xml
+      **/**/target/surefire-reports/TEST*.xml
+      **/**/**/target/surefire-reports/TEST*.xml
+      testsuite/integration-arquillian/tests/base/target/surefire-reports/TEST*.xml
+      quarkus/tests/integration/target/surefire-reports/TEST*.xml
+      testsuite/model/target/surefire-reports*/TEST*.xml
+      crypto/default/target/surefire-reports/TEST*.xml
+      crypto/fips1402/target/surefire-reports/TEST*.xml
+      testsuite/integration-arquillian/tests/other/base-ui/target/surefire-reports/TEST*.xml
+      testsuite/integration-arquillian/tests/other/webauthn/target/surefire-reports/TEST*.xml
+      testsuite/integration-arquillian/tests/other/sssd/target/surefire-reports/TEST*.xml
+  keep-days:
+    description: 'For how many days to store the particular artifact.'
+    required: false
+    default: 1
+
+runs:
+  using: composite
+  steps:
+    - id: upload-surefire
+      name: Upload Surefire reports
+      if: (!cancelled() && contains(fromJSON(inputs.release-branches), github.ref) && contains(fromJSON('["push", "workflow_dispatch"]'), github.event_name))
```

Comment:
@stianst @miquelsi Do you think it's relevant to collect also the results from failed builds? Or are we interested in success once? thanks

Reply:
I think we want from failed builds as well, otherwise we would not upload anything if there are test failures.